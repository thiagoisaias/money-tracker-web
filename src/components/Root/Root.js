import React from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import PropTypes from "prop-types";

import App from "../../containers/App/App";

const Root = props => {
  return (
    <Provider store={props.store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
};

Root.propTypes = {
  store: PropTypes.object
};

export default Root;
