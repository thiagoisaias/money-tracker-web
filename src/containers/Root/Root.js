import React, { Component, Fragment } from "react";
import { Redirect, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { checkAuthStorage } from "../../store/actions/auth";

import App from "../../components/App/App";

export class Root extends Component {
  render() {
    let authRedirect = null;
    if (
      !this.props.isAuthenticated &&
      this.props.location.pathname !== "/auth"
    ) {
      authRedirect = <Redirect to="/auth" />;
    }
    return (
      <Fragment>
        {authRedirect}
        <App />
      </Fragment>
    );
  }

  componentDidMount() {
    this.props.checkAuthStorage();
  }
}

Root.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  checkAuthStorage: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    isAuthenticated: state.auth.isAuthenticated
  };
};

const mapDispatchToProps = dispatch => {
  return {
    checkAuthStorage: () => {
      dispatch(checkAuthStorage());
    }
  };
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Root));
