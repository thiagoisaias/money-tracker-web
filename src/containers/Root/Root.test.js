import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Root } from "./Root";

describe("Root", () => {
  const mockProps = {
    isAuthenticated: true,
    checkAuthStorage: jest.fn()
  };
  const component = shallow(<Root {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
