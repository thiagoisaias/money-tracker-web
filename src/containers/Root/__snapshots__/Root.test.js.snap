// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders properly 1`] = `
<React.Fragment>
  <App />
</React.Fragment>
`;
